### PR TITLE
added domclust

### PR DESCRIPTION
**Pull request by PatrickRWright**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by eggzilla** [comment]
Nice :-) Plz also add the LICENSE

File: recipes/domclust/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  license: 
+  summary: Effective tool for orthologous grouping in multiple genomes 
+
+build:
+  number: 0
+  skip: True # [osx]
+
+source:
+  fn: domclust.tgz 
+  url: "http://mbgd.genome.ad.jp/domclust/domclust.tgz"
```
Comment (bgruening) [inline recipes/domclust/meta.yaml]: This does not look like a versioned URL.  If you do not trust this URL please mirror it here with https://github.com/galaxyproject/cargo-port

File: recipes/domclust/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# copy binary to prefix folder
+cp -p bin/domclust $PREFIX/bin
+cp -p Script/addtit.pl $PREFIX/bin
+cp -p Script/cmpr.pl $PREFIX/bin
+cp -p Script/convgraph.pl $PREFIX/bin
+cp -p build_input/blast2homfile.pl $PREFIX/bin
+cp -p build_input/fasta2genefile.pl $PREFIX/bin
+
+# copy test data
+cp -p tst/test.gene $PREFIX/share
```
Comment (bgruening) [inline recipes/domclust/build.sh]: How big are those files? Please note that we just check if the installation succeeded and not run functional test in BioConda.

File: recipes/domclust/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package:
+  name: domclust 
+  version: "1.2.7l"
+
+about:
+  home: http://mbgd.genome.ad.jp/domclust/
+  license:
```
Comment (bgruening) [inline recipes/domclust/meta.yaml]: license missing

File: recipes/domclust/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  fn: domclust.tgz 
+  url: "http://mbgd.genome.ad.jp/domclust/domclust.tgz"
+  sha256: 9e938027f2a0b859feaea5a7701c9b7cd41d494a49cb7bd66890077dec551e3e 
+
+requirements:
+  build:
+    # linux build
+    - gcc # [linux]
+    - perl
+    # osx build
```
Comment (bgruening) [inline recipes/domclust/meta.yaml]: can you remove these comments, if they are not needed?

File: recipes/domclust/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    # linux build
+    - libgcc # [linux]
+    #- libcxx # [osx]
+    - perl
+    #- boost ==1.63.0 # [osx]
+
+test:
+  commands:
+    - domclust -h
+    - domclust $PREFIX/share/test.hom $PREFIX/share/test.gene
```
Comment (bgruening) [inline recipes/domclust/meta.yaml]: PREFIX does not exists in the test data section (in Docker). You could:
- skip this test
- use a hack like `which domclust`/../share/test.hom
- run the test as part of the build.sh script

Comment (bgruening) [inline recipes/domclust/meta.yaml]: My bad, I implemented support for PREFIX already. Leave it like it is.

File: recipes/domclust/meta.yaml (hunk 3 above)
Comment (PatrickRWright) [inline recipes/domclust/meta.yaml]: As far as I can see the original developers have not supplied one. What is the best practice in such a case?

File: recipes/domclust/meta.yaml (hunk 1 above)
Comment (PatrickRWright) [inline recipes/domclust/meta.yaml]: I agree. However, given the licensing issue above I am wondering if it is legit to just mirror this!?

File: recipes/domclust/build.sh (hunk 2 above)
Comment (PatrickRWright) [inline recipes/domclust/build.sh]: 201K, 301K and 5.2M pretty small...

File: recipes/domclust/meta.yaml (hunk 3 above)
Comment (bgruening) [inline recipes/domclust/meta.yaml]: If there is no license you are not allowed to do anything with it :(
Can you ask the authors?

File: recipes/domclust/build.sh (hunk 2 above)
Comment (bgruening) [inline recipes/domclust/build.sh]: I would not put this into the package. You can run the tests inside the build.sh script and then remove the test data if you like.

File: recipes/domclust/meta.yaml (hunk 3 above)
Comment (PatrickRWright) [inline recipes/domclust/meta.yaml]: I will ask.

File: recipes/domclust/meta.yaml [hunk 6]
```diff
@@ -1,27 +1,24 @@
...
+  url: "https://depot.galaxyproject.org/software/DomClust/DomClust_1.2.8_src_all.tar.gz"
+  sha256: ffe3dac8dee1ccd621556ed7e5818299ed167853dd4dcd9e42cde43eec4c40b2 
 
 requirements:
   build:
-    # linux build
     - gcc # [linux]
     - perl
-    # osx build
     - llvmdev ==3.9.1 # [osx]
```
Comment (bgruening) [inline recipes/domclust/meta.yaml]: can you change this to just `- llvm`

Comment (PatrickRWright) [inline recipes/domclust/meta.yaml]: sure

**Pull request merged**